presto-main/src/test/java/com/facebook/presto/operator/TestUncheckedByteArrays.java

```diff
@@ -0,0 +1,228 @@
+        int filledBytes = fill(destination, 0, POSITIONS_PER_PAGE, (byte) 5);
+
+        assertEquals(filledBytes, POSITIONS_PER_PAGE);
+
+        ArrayList<Byte> expected = new ArrayList<Byte>(Collections.nCopies(POSITIONS_PER_PAGE, (byte) 5));
+        assertCopied(expected, destination, ARRAY_BYTE_INDEX_SCALE, (bytes, index) -> getByte((byte[]) bytes, (int) index));
+    }
+
+    @Test
+    public static void testSetBytes()
+    {
+        final byte[] destination = new byte[POSITIONS_PER_PAGE];
+
+        byte[] source = new byte[POSITIONS_PER_PAGE];
+        RANDOM.nextBytes(source);
+
+        int setBytes = setBytes(destination, 0, source, 0, POSITIONS_PER_PAGE);
+
+        assertEquals(setBytes, POSITIONS_PER_PAGE);
+
+        assertCopied(Bytes.asList(source), destination, ARRAY_BYTE_INDEX_SCALE, (bytes, index) -> getByte((byte[]) bytes, (int) index));
+    }
+
+    @Test
+    public static void testSetInts()
+    {
+        final byte[] destination = new byte[POSITIONS_PER_PAGE * ARRAY_INT_INDEX_SCALE];
+
+        int copiedBytes = setInts(destination, 0, IntStream.range(0, POSITIONS_PER_PAGE).toArray(), 0, POSITIONS_PER_PAGE);
+
+        assertEquals(copiedBytes, POSITIONS_PER_PAGE * ARRAY_INT_INDEX_SCALE);
+
+        List<Integer> expected = IntStream.range(0, POSITIONS_PER_PAGE).boxed().collect(Collectors.toList());
+        assertCopied(expected, destination, ARRAY_INT_INDEX_SCALE, (bytes, index) -> getInt((byte[]) bytes, (int) index));
+    }
+
+    private static <T> void testSequential(List<T> source, int elementSize, byte[] destination, TriFunction writeFunction, BiFunction readFunction)
+    {
+        int index = 0;
+        for (int i = 0; i < source.size(); i++) {
+            index = writeFunction.apply(destination, index, (T) source.get(i));
+        }
+        assertEquals(index, POSITIONS_PER_PAGE * elementSize);
+        assertCopied(source, destination, elementSize, readFunction);
+    }
+
+    // source should contain identical elements.
+    private static <T> void testRandom(List<T> source, int elementSize, byte[] destination, TriFunction writeFunction, BiFunction readFunction)
+    {
+        Arrays.fill(destination, (byte) 0);
+
+        int index = 0;
+        for (int i = 0; i < source.size(); i++) {
+            int newIndex = writeFunction.apply(destination, index, (T) source.get(i));
+            if (i % 2 == 0) {
+                index = newIndex;
+            }
+        }
+
+        assertEquals(index, POSITIONS_PER_PAGE / 2 * elementSize);
+        assertCopied(source.subList(0, POSITIONS_PER_PAGE / 2), Arrays.copyOf(destination, POSITIONS_PER_PAGE / 2), elementSize, readFunction);
+        assertCopied(source.subList(POSITIONS_PER_PAGE / 2 + 1, POSITIONS_PER_PAGE), Arrays.copyOf(destination, POSITIONS_PER_PAGE / 2), elementSize, readFunction);
+    }
+
+    private static <T> boolean assertCopied(List<T> expected, byte[] actual, int elementSize, BiFunction readFunction)
```

Comment:
- this method must assert the condition and not return a boolean
- the use of `index` variable below is a suspect... (the test fails after introducing the assert)... could you double check?

```
    private static <T> void assertCopied(List<T> expected, byte[] actual, int elementSize, BiFunction<byte[], Integer, T> readFunction)
    {
        for (int index = 0; index < actual.length; index++) {
            if (readFunction.apply(actual, index) != expected.get(index)) {
                fail("Value at index " + index + " doesn't match expected value of " + expected.get(index));
            }
            index += elementSize;
        }
    }
```